pkg/controller/vm/controller.go

```diff
@@ -56,12 +58,21 @@ func (h *Handler) OnChange(key string, vm *kubevirtv1.VirtualMachine) (*kubevirt
 	ncm := make(map[string]networkv1.NetworkConfig, 1)
 
 	// Construct initial network config map
-	for _, nic := range vm.Spec.Template.Spec.Domain.Devices.Interfaces {
-		if nic.MacAddress == "" {
+	nicToMacAddress := map[string]string{}
```

Comment:
we should probably also iterate over interfaces in `vm.Spec.Template.Spec.Domain.Devices` and check if there is a mac address present and update the map if a non empty mac address string is found since that will be eventually using in the domain definition.

it will also ensure that if a user specifies a mac address, specially like via hot plug nic operations the mac address from the domain spec is used.

Reply:
@ibrokethecloud I addressed both comments. Thank you.